mindmaps-migration/sql/src/main/java/io/mindmaps/migration/sql/SQLMigrator.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * MindmapsDB - A Distributed Semantic Database
+ * Copyright (C) 2016  Mindmaps Research Ltd
+ *
+ * MindmapsDB is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MindmapsDB is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MindmapsDB. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package io.mindmaps.migration.sql;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.Maps;
+import io.mindmaps.graql.InsertQuery;
+import io.mindmaps.migration.base.AbstractMigrator;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.impl.DSL;
+
+import java.sql.Connection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * The SQL migrator will execute the given SQL query and then apply the given template to those results.
+ */
+public class SQLMigrator extends AbstractMigrator {
+
+    private final Stream<Record> records;
```

Comment:
Because streams are single use, calling `migrate` twice will throw an exception. This might be the desired behaviour though!